golem-worker-executor-base/tests/api.rs

```diff
@@ -2156,29 +2183,51 @@ async fn long_running_poll_loop_connection_can_be_restored_after_resume(
         .await;
     let mut rx = executor.capture_output_with_termination(&worker_id).await;
 
+    // wait for one loop to finish
+    {
+        let mut found1 = false;
+        let mut found2 = false;
+        while !(found1 && found2) {
+            match rx.recv().await {
+                Some(Some(event)) => {
+                    if stdout_event_matching(&event, "Calling the poll endpoint\n") {
+                        found1 = true;
+                    } else if stdout_event_matching(&event, "Received initial\n") {
+                        found2 = true;
+                    }
+                }
+                _ => {
+                    panic!("Did not receive the expected log events");
+                }
+            }
+        }
+    }
+
+    // change the response. Next loop will be the last
     {
         let mut response = response.lock().unwrap();
         *response = "first".to_string();
     }
 
-    let start = Instant::now();
```

Comment:
If we had bad luck the interleaving was like this, causing the test to timeout

log("Calling the poll endpoint\n") *> connect *> change_response *> log("Received first\n") *> log("Poll loop finished\n")